clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -208,16 +208,28 @@ static bool isZeroSizedArray(QualType Ty) {
 
 static Sema::DeviceDiagBuilder
 emitDeferredDiagnosticAndNote(Sema &S, SourceRange Loc, unsigned DiagID,
-                              SourceRange UsedAtLoc) {
-  Sema::DeviceDiagBuilder builder =
-      S.SYCLDiagIfDeviceCode(Loc.getBegin(), DiagID);
-  if (UsedAtLoc.isValid())
-    S.SYCLDiagIfDeviceCode(UsedAtLoc.getBegin(), diag::note_sycl_used_here);
+                              bool DiagImmediately, SourceRange UsedAtLoc) {
+  FunctionDecl *FD = dyn_cast<FunctionDecl>(S.getCurLexicalContext());
+  auto builder =
+      DiagImmediately
+          ? Sema::DeviceDiagBuilder(Sema::DeviceDiagBuilder::K_Immediate,
+                                    Loc.getBegin(), DiagID, FD, S)
+          : S.SYCLDiagIfDeviceCode(Loc.getBegin(), DiagID);
+  if (UsedAtLoc.isValid()) {
+    if (DiagImmediately)
+      Sema::DeviceDiagBuilder(Sema::DeviceDiagBuilder::K_Immediate,
+                              UsedAtLoc.getBegin(), diag::note_sycl_used_here,
+                              FD, S);
+    else
+      S.SYCLDiagIfDeviceCode(UsedAtLoc.getBegin(), diag::note_sycl_used_here);
```

Comment:
Wow, we really have `note_sycl_used_here` note which has exact the same text as default `note_used_here`. I have no idea hot this happened, but please, let's remove it and don't use it further.
BTW, when it should be emitted? I haven't seen any test cases emitting it...

Reply:
you'll see test cases in sycl-restrict.cpp which is where most of the type check tests occur. That note is emitted when struct members have illegal types.  

I'll switch it over to `note_used_here`.

Reply:
done